### PR TITLE
Replace checkboxes with Material Component switches.

### PR DESCRIPTION
**Pull request by BugsNash**
Part of #369

Default/dark theme before:

screenshot from 2018-12-19 15-09-47

Default/dark theme after:

screenshot from 2018-12-19 15-12-00

Light theme before:

screenshot from 2018-12-19 15-10-18

Light theme after:

screenshot from 2018-12-19 15-10-31

### CODE REVIEW

File: js/controllers/settings.js [hunk 1]
```diff
@@ -60,6 +60,28 @@ SettingsController.prototype.show_ = function(key, value) {
   }
 };
 
+/**
+ * Activates a switch Material Component element in the UI.
+ * @param {string} key The unique section of the id of the switch element
+ *     (after the 'setting-' prefix).
```
Comment (E-T-) [inline js/controllers/settings.js]: Add `@private` to both this and the other method.

File: js/controllers/settings.js [hunk 2]
```diff
@@ -60,6 +60,28 @@ SettingsController.prototype.show_ = function(key, value) {
...
+ */
+SettingsController.prototype.checkSwitch_ = function(key) {
+  document.getElementById('setting-' + key).setAttribute('checked', '');
```
Comment (E-T-) [inline js/controllers/settings.js]: Can you use https://developer.mozilla.org/en-US/docs/Web/API/Element/toggleAttribute? I think there's also a `toggleClass` on classlist too.

Comment (BugsNash) [reply js/controllers/settings.js]: Currently this code is used to do more than just toggle, sometimes we are setting to false when it's already false (e.g. in initial setup of all the settings switches). Do you want me to change the setup code to instead check the value and conditionally toggle if the initial value is supposed to be true?

Comment (BugsNash) [reply js/controllers/settings.js]: oh actually I can just use the second argument of the toggle function to state whether the switch should be turned on or off.

File: js/controllers/settings.js (hunk 1 above)
Comment (BugsNash) [reply js/controllers/settings.js]: Done.

**Pull request merged**